pkg/injection/namespace/mapper/namespaces.go

```diff
@@ -19,21 +21,27 @@ type NamespaceMapper struct {
 }
 
 func NewNamespaceMapper(ctx context.Context, clt client.Client, apiReader client.Reader, operatorNs string, targetNs *corev1.Namespace) NamespaceMapper { //nolint:revive // argument-limit doesn't apply to constructors
-	return NamespaceMapper{ctx, clt, apiReader, operatorNs, targetNs}
+	return NamespaceMapper{ctx, controller_runtime.NewClient(clt), apiReader, operatorNs, targetNs}
 }
 
 // MapFromNamespace adds the labels to the targetNs if there is a matching Dynakube
-func (nm NamespaceMapper) MapFromNamespace() (bool, error) {
-	updatedNamespace, err := nm.updateNamespace()
+func (nm NamespaceMapper) MapFromNamespace(ctx context.Context) (bool, error) {
+	ctx, span := injectionotel.StartSpan(ctx, "NamespaceMapper.MapFromNamespace")
+	defer span.End()
+
+	updatedNamespace, err := nm.updateNamespace(ctx)
 	if err != nil {
 		return false, err
 	}
 	return updatedNamespace, nil
 }
 
-func (nm NamespaceMapper) updateNamespace() (bool, error) {
+func (nm NamespaceMapper) updateNamespace(ctx context.Context) (bool, error) {
+	ctx, span := injectionotel.StartSpan(ctx, "NamespaceMapper.updateNamespace")
+	defer span.End()
+
 	deployedDynakubes := &dynatracev1beta1.DynaKubeList{}
-	err := nm.client.List(nm.ctx, deployedDynakubes)
+	err := nm.client.List(ctx, deployedDynakubes)
```

Comment:
line 47 - maybe we should report an error
```
  span.ReportError(err)
  return
```

Reply:
good point, added error recording.